src/evergreen/evg_config_utils.rs

```diff
@@ -517,6 +534,61 @@ impl EvgConfigUtils for EvgConfigUtilsImpl {
             .collect()
     }
 
+    /// Determine burn in tag build variants for the given build variant.
+    ///
+    /// # Arguments
+    ///
+    /// * `build_variant` - Build Variant to query.
+    /// * `build_variant_map` - A map of build variant names to their definitions.
+    ///
+    /// # Returns
+    ///
+    /// List of build variant names to use for burn in tags.
+    fn resolve_burn_in_tag_build_variants(
+        &self,
+        build_variant: &BuildVariant,
+        build_variant_map: &HashMap<String, &BuildVariant>,
+    ) -> Vec<String> {
+        let mut burn_in_build_variants = self
+            .lookup_and_split_by_whitespace_build_variant_expansion(
+                BURN_IN_TAG_INCLUDE_BUILD_VARIANTS,
+                build_variant,
+            );
+        if self
+            .lookup_build_variant_expansion(
+                BURN_IN_TAG_REQUIRED_AND_SUGGESTED_BUILD_VARIANTS,
+                build_variant,
+            )
+            .unwrap_or_default()
+            == "true"
```

Comment:
Comparing string value with "true" seems a bit sketchy. I'm not super familiar with type conversion in rust, maybe [this](https://doc.rust-lang.org/std/string/struct.String.html#method.parse) can help.